### PR TITLE
[CI] Add acceptance comparison workflow (target vs baseline)

### PR DESCRIPTION
**Pull request by mengfei25**
## Acceptance comparison workflow and CI enhancements

Add infrastructure for running target-vs-baseline acceptance comparisons for triton and oneDNN components.

### New files

- **`acceptance_ondemand.yml`** — orchestration workflow that builds and tests two pipelines (target vs baseline) on the same runner, then compares results
  - Input model: pick `test_component` (triton or onednn) and set `target_version` / `baseline_version`; common `pytorch` and `torch_xpu_ops` base refs shared by both sides
  - Fixed test scope: `p0` UT + E2E (huggingface, timm_models, torchbench — training amp_fp16 + inference bfloat16, accuracy + performance)
  - Comparison steps use `continue-on-error` so partial failures still produce summaries
- **`compare-e2e.py`** — E2E result comparison script (accuracy + performance, xlsx + markdown output); uses `pd.to_numeric` coercion for safe arithmetic
- **`compare-ut.py`** — UT result comparison script (pass/fail diffs, xlsx + markdown output); checks `gh` subprocess return codes
- **`.ci/benchmarks/p0/run_ut.sh`** — P0 unit test runner script

### Workflow changes

**`_linux_build.yml`**
- Added `onednn` input — override XPU oneDNN ref (`commit/branch/tag` or `repo@ref`; empty/pinned = pytorch default)
- Added `category` input — tag (e.g. `target`/`baseline`) appended to wheel/log artifact names to avoid collisions
- Passes `ONEDNN_COMMIT` env var to `build.sh`

**`_linux_e2e.yml`**
- Added `category` input, propagated to test-env setup and artifact upload names
- Added `build-comparison` test type with fixed scope: training amp_fp16 + inference bfloat16, accuracy + performance

**`_linux_ut.yml`**
- Added `category` input, propagated to test-env setup and all artifact upload names
- Added `skip_summary` boolean input (decoupled from `category`) to suppress the summary/GitHub-issue job when the caller handles its own comparison

**`linux-testenv/action.yml`**
- Added `category` input to select the matching wheel artifact pattern (`Torch-XPU-Wheel-*-{category}`)

**`linux-uttest/action.yml`**
- Added `p0` test step (runs `.ci/benchmarks/p0/run_ut.sh`, 20h timeout)
- Added `if: !cancelled()` guard on the UT summary step

### Build script changes

**`build.sh`**
- Added `ONEDNN_COMMIT` override — patches `GIT_REPOSITORY`/`GIT_TAG` in `cmake/Modules/FindMKLDNN.cmake` via sed
- Supports `commit/tag` or `repo@ref` format; `pinned` or empty = no-op

### CODE REVIEW

**Comment by mengfei25** [comment]
Run passed manually https://github.com/mengfei25/torch-xpu-ops/actions/runs/25775592491

File: .github/actions/linux-uttest/action.yml [hunk 1]
```diff
@@ -155,6 +155,25 @@ runs:
           tee ${{ github.workspace }}/ut_log/xpu_distributed/xpu_distributed_test.log
         find ../ -type f -name "*.xml" -exec cp {} ${{ github.workspace }}/ut_log/ \;
 
+    - name: op_p0
```
Comment (chuanqi129) [inline .github/actions/linux-uttest/action.yml]: how about rename to `p0`?

File: .github/scripts/build.sh [hunk 2]
```diff
@@ -46,6 +46,47 @@ cd ${WORKSPACE}/pytorch
 python -m pip install requests
 python third_party/torch-xpu-ops/.github/scripts/apply_torch_pr.py
 git submodule sync && git submodule update --init --recursive
+
+# Optional component overrides (used by the acceptance / comparison workflow).
+# Each variable, when set to a non-empty value other than "pinned",
+# replaces the default pinned version of the corresponding component.
+
+# oneDNN: pytorch's mkl-dnn submodule lives under third_party/ideep/mkl-dnn
```
Comment (chuanqi129) [inline .github/scripts/build.sh]: It's not true, xpu part live in the cmake file, we should update the cmake file. 2 possible methods, update the commit to what we want to test in the file, or delete the git clone part in the cmake file and prepare oneDNN source code by ourselves and copy it to the destination folder

File: .github/scripts/build.sh [hunk 3]
```diff
@@ -46,6 +46,47 @@ cd ${WORKSPACE}/pytorch
...
+    # Try fetching the ref directly (works for branch/tag/SHA on servers
+    # with uploadpack.allowReachableSHA1InWant); fall back to a full fetch.
+    git fetch --tags --force origin "${ONEDNN_REF}" || git fetch --tags --force origin
+    git checkout --force "${ONEDNN_REF}"
+    git submodule update --init --recursive
+    git rev-parse HEAD
+    popd >/dev/null
+fi
+
+# SYCLTLA: cmake FetchContent in torch-xpu-ops/cmake/SYCLTLA.cmake
```
Comment (chuanqi129) [inline .github/scripts/build.sh]: SYCLTLA don't have request for now, let's clean this part

File: .github/workflows/acceptance_ondemand.yml [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+        type: string
+        default: ''
+        description: 'Target oneDNN ref. Empty/`pinned` = pytorch-pinned.'
+      target_sycltla:
+        type: string
+        default: ''
+        description: 'Target SYCLTLA ref. Empty/`pinned` = torch-xpu-ops-pinned.'
+
+      # ----- Baseline versions -----
+      baseline_pytorch:
```
Comment (chuanqi129) [inline .github/workflows/acceptance_ondemand.yml]: I think for the acceptance test, we just need component target and baseline, others are keep same, that means pytorch, torch-xpu-ops should same. And suggest to define 3 inputs, test_component (onednn, triton, ...), target version, baseline version

File: .github/actions/linux-uttest/action.yml (hunk 1 above)
Comment (mengfei25) [reply .github/actions/linux-uttest/action.yml]: Done

File: .github/scripts/build.sh (hunk 2 above)
Comment (mengfei25) [reply .github/scripts/build.sh]: Done

File: .github/scripts/build.sh (hunk 3 above)
Comment (mengfei25) [reply .github/scripts/build.sh]: Done

File: .github/workflows/acceptance_ondemand.yml (hunk 4 above)
Comment (mengfei25) [reply .github/workflows/acceptance_ondemand.yml]: Done

File: .github/workflows/_linux_e2e.yml [hunk 5]
```diff
@@ -224,6 +229,24 @@ jobs:
           dt: ${{ steps.dtypes.outputs.pt2e_dtypes }}
           scenario: ${{ inputs.scenario }}
 
+      # Acceptance comparison launch (fixed scope: training amp_fp16 + inference bfloat16, accuracy+performance)
+      - name: Comparison Test - training (${{ inputs.suite }})
```
Comment (chuanqi129) [inline .github/workflows/_linux_e2e.yml]: Where we limit the e2e test model list to p0 scope?

File: .github/workflows/acceptance_ondemand.yml [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+# (`pytorch`, `torch_xpu_ops`, `triton`, `onednn`) via `test_component`,
+# then set `target_version` / `baseline_version` for that component.
+# Other components use their base inputs or upstream-pinned defaults.
+
+on:
+  workflow_dispatch:
+    inputs:
+      runner:
+        type: string
+        default: pvc_rolling
```
Comment (chuanqi129) [inline .github/workflows/acceptance_ondemand.yml]: Do we have a dedicated runners for this tests?

File: .github/workflows/acceptance_ondemand.yml [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    secrets: inherit
+    uses: ./.github/workflows/_linux_build.yml
+    with:
+      runner: build
+      pytorch: ${{ inputs.test_component == 'pytorch' && inputs.target_version || inputs.pytorch }}
+      torch_xpu_ops: ${{ inputs.test_component == 'torch_xpu_ops' && inputs.target_version || inputs.torch_xpu_ops }}
+      triton: ${{ inputs.test_component == 'triton' && inputs.target_version || '' }}
+      onednn: ${{ inputs.test_component == 'onednn' && inputs.target_version || '' }}
+      python: ${{ inputs.python }}
+      category: target
```
Comment (chuanqi129) [inline .github/workflows/acceptance_ondemand.yml:91]: Please launch a test to verify the wheel whether use that passed oneDNN commit

File: .github/workflows/acceptance_ondemand.yml [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+        description: 'Base PyTorch ref (commit/branch/tag, or `repo@ref`). Overridden when test_component=pytorch.'
+      torch_xpu_ops:
+        type: string
+        default: 'main'
+        description: 'Base torch-xpu-ops ref, or `pinned`. Overridden when test_component=torch_xpu_ops.'
+
+      # ----- Component under test -----
+      test_component:
+        type: choice
+        options: [onednn, triton, torch_xpu_ops, pytorch]
```
Comment (chuanqi129) [inline .github/workflows/acceptance_ondemand.yml]: Why we need torch-xpu-ops & pytorch here, keep oneDNN triton.

File: .github/workflows/_linux_e2e.yml (hunk 5 above)
Comment (mengfei25) [reply .github/workflows/_linux_e2e.yml]: in _linux_e2e.yml and also commented it in PR desc

File: .github/workflows/acceptance_ondemand.yml (hunk 6 above)
Comment (mengfei25) [reply .github/workflows/acceptance_ondemand.yml]: added a dedicated runner `acceptance` label for this tests

File: .github/workflows/acceptance_ondemand.yml (hunk 8 above)
Comment (mengfei25) [reply .github/workflows/acceptance_ondemand.yml]: removed

File: .github/workflows/_linux_e2e.yml
Comment (chuanqi129) [inline .github/workflows/_linux_e2e.yml:13]: add `build-acceptance` in here

File: .github/workflows/_linux_e2e.yml [hunk 9]
```diff
@@ -224,6 +229,24 @@ jobs:
...
+        if: ${{ contains(inputs.test_type, 'comparison') }}
```
Comment (chuanqi129) [inline .github/workflows/_linux_e2e.yml]:
```suggestion
        if: ${{ contains(inputs.test_type, 'acceptance') }}
```

File: .github/workflows/_linux_e2e.yml [hunk 10]
```diff
@@ -224,6 +229,24 @@ jobs:
...
+        uses: ./.github/actions/linux-e2etest
+        with:
```
Comment (chuanqi129) [inline .github/workflows/_linux_e2e.yml:239]: those 2 steps defined the test configs but use full model list. Am I right?

Comment (mengfei25) [reply .github/workflows/_linux_e2e.yml:239]: Yes, use pytorch defined model list as p0 scope.

File: .github/workflows/_linux_e2e.yml
Comment (mengfei25) [reply .github/workflows/_linux_e2e.yml:13]: done

File: .github/workflows/_linux_e2e.yml (hunk 9 above)
Comment (mengfei25) [reply .github/workflows/_linux_e2e.yml]: done

File: .github/workflows/acceptance_ondemand.yml (hunk 7 above)
Comment (mengfei25) [reply .github/workflows/acceptance_ondemand.yml:91]: effective
https://github.com/mengfei25/torch-xpu-ops/actions/runs/25846038446

**Comment by mengfei25** [comment]
Acceptance functional: https://github.com/mengfei25/torch-xpu-ops/actions/runs/25846038446
On-demand functional: https://github.com/mengfei25/torch-xpu-ops/actions/runs/25775682291

**Review by chuanqi129 — APPROVED** [review]
